### DISCUSSION TITLE
SMTP configuration in Overlay File for Micrososft OAuth2

### DISCUSSION THREAD

**Discussion by kartraj** [hystax/optscale#129] [General]
@maxb-hystax @ek-hystax 
We are attempting to setup the SMTP server to send emails from the Optscale application. The overlay files have the following configurations:

smtp:
  server:
  email:
  port:
  password:

But we do not want to use a hard-coded username and password. instead, we want to use the Microsoft OAuth method, which requires an access token to be generated. Is it possible to configure the SMTP for Optscale through this method? How would the configurations in the overlay file change based on this?

Ref: [Authenticate an IMAP, POP or SMTP connection using OAuth | Microsoft Learn]

https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#authenticate-connection-requests

**Comment by maxb-hystax** [comment]
Hi Karthik,

Current implementation works only with login/password, see https://github.com/hystax/optscale/blob/integration/herald/modules/email_sender/sender.py
